### PR TITLE
docs: Add documentation for "hugo deploy"

### PR DESCRIPTION
**Pull request by vangent**
Updates #5943.

### CODE REVIEW

File: docs/content/en/hosting-and-deployment/hugo-deploy.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+[deployment]
+# By default, files are uploaded in an arbitrary order.
+# Files that match the regular expressions in the "Order" list
+# will be uploaded first, in the listed order.
+Order = [".jpg$", ".gif$"]
+
+
+[[deployment.targets]]
+# An arbitrary name for this target.
+Name = "mydeployment"
```
[the remark this answers is not in the document]
Comment (vangent) [reply docs/content/en/hosting-and-deployment/hugo-deploy.md]: Done.

File: docs/content/en/hosting-and-deployment/hugo-deploy.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# ... add more [[deployment.targets]] sections ...
+
+
+# [[deployment.matchers]] configure behavior for files that match the Pattern.
+# Samples:
+
+[[deployment.matchers]]
+#  Cache static assets for 20 years.
+Pattern = "^.+\\.(js|css|svg|ttf)$"
+Cache-Control = "max-age=630720000, no-transform, public"
```
[the remark this answers is not in the document]
Comment (vangent) [reply docs/content/en/hosting-and-deployment/hugo-deploy.md]: Done. This required removing the `mapstructure` tag in the code, so I did that, and updated the tests.

File: docs/content/en/hosting-and-deployment/hugo-deploy.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (vangent) [reply docs/content/en/hosting-and-deployment/hugo-deploy.md]: OK -- they are case-insensitive when processed, so it doesn't really matter, but updated the examples.

I left `URL` all caps though; let me know if you think `url` would be more consistent.

File: docs/content/en/hosting-and-deployment/hugo-deploy.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+[[deployment.matchers]]
+Pattern = "^.+\\.(html|xml|json)$"
+gzip = true
+```
+
+## Deploy
+
+To deploy to a target:
+```
+hugo deploy --target=<target>
```
[the remark this answers is not in the document]
Comment (vangent) [reply docs/content/en/hosting-and-deployment/hugo-deploy.md]: Currently it fails. I filed https://github.com/gohugoio/hugo/issues/6008 to default to the first target.

File: docs/content/en/hosting-and-deployment/hugo-deploy.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+Hugo will identify any local changes that need to be uploaded, and ask for
+confirmation before doing anything.
```
[the remark this answers is not in the document]
Comment (vangent) [reply docs/content/en/hosting-and-deployment/hugo-deploy.md]: You are right; updated.

File: docs/content/en/hosting-and-deployment/hugo-deploy.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bep) [reply docs/content/en/hosting-and-deployment/hugo-deploy.md]: `URL` is good. We kind of follow Go's naming here, so it`s `baseURL`, `fooID` etc.

**Comment by bep** [comment]
Looks good. Can you fix the commit messages (the last 2) and I'll merge it?

**Comment by vangent** [comment]
> Looks good. Can you fix the commit messages (the last 2) and I'll merge it?

Sorry, do you mean squash and force-push?

**Comment by bep** [comment]
No, I mean the commit messages (start with a capital letter).

**Comment by vangent** [comment]
> No, I mean the commit messages (start with a capital letter).

Done.

**Pull request merged**